gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -17,202 +17,156 @@
-    this.streamStatistics = ghfs.getInstrumentation().newInputStreamStatistics(statistics);
-    this.ioStatistics = GoogleHadoopFSInputStream.this.streamStatistics.getIOStatistics();
+    logger.atFiner().log("create(gcsPath: %s)", gcsPath);
+    GoogleCloudStorageFileSystem gcsFs = ghfs.getGcsFs();
+    SeekableByteChannel channel =
+        gcsFs.open(gcsPath, gcsFs.getOptions().getCloudStorageOptions().getReadChannelOptions());
+    return new GoogleHadoopFSInputStream(ghfs, gcsPath, channel, statistics);
   }
 
-  GoogleHadoopFSInputStream(
+  static GoogleHadoopFSInputStream create(
       GoogleHadoopFileSystem ghfs, FileInfo fileInfo, FileSystem.Statistics statistics)
       throws IOException {
-    logger.atFiner().log("GoogleHadoopFSInputStream(fileInfo: %s)", fileInfo);
-    this.gcsPath = fileInfo.getPath();
-    this.statistics = statistics;
-    this.totalBytesRead = 0;
+    logger.atFiner().log("create(fileInfo: %s)", fileInfo);
     GoogleCloudStorageFileSystem gcsFs = ghfs.getGcsFs();
-    this.channel =
+    SeekableByteChannel channel =
         gcsFs.open(fileInfo, gcsFs.getOptions().getCloudStorageOptions().getReadChannelOptions());
+    return new GoogleHadoopFSInputStream(ghfs, fileInfo.getPath(), channel, statistics);
+  }
+
+  private GoogleHadoopFSInputStream(
+      GoogleHadoopFileSystem ghfs,
+      URI gcsPath,
+      SeekableByteChannel channel,
+      FileSystem.Statistics statistics) {
+    logger.atFiner().log("GoogleHadoopFSInputStream(gcsPath: %s)", gcsPath);
+    this.gcsPath = gcsPath;
+    this.channel = channel;
+    this.statistics = statistics;
     this.streamStatistics = ghfs.getInstrumentation().newInputStreamStatistics(statistics);
-    this.ioStatistics = GoogleHadoopFSInputStream.this.streamStatistics.getIOStatistics();
   }
 
   @Override
   public synchronized int read() throws IOException {
-    streamStatistics.readOperationStarted(getPos(), 1);
+    streamStatistics.readOperationStarted(getPos(), /* len= */ 1);
     int response;
     try {
-      int result;
       // TODO(user): Wrap this in a while-loop if we ever introduce a non-blocking mode for the
       // underlying channel.
-
-      synchronized (this) {
-        int numRead = channel.read(ByteBuffer.wrap(singleReadBuf));
-        if (numRead == -1) {
-          result = -1;
-        } else {
-          if (numRead != 1) {
-            throw new IOException(
-                String.format(
-                    "Somehow read %d bytes using single-byte buffer for path %s ending in position %d!",
-                    numRead, gcsPath, channel.position()));
-          }
-          byte b = singleReadBuf[0];
-          totalBytesRead++;
-          statistics.incrementBytesRead(1);
-          statistics.incrementReadOps(1);
-          result = (b & 0xff);
+      int numRead = channel.read(ByteBuffer.wrap(singleReadBuf));
+      if (numRead == -1) {
+        response = -1;
+      } else {
+        if (numRead != 1) {
```

Comment:
Can we collapse with ` else if ` instead

Reply:
Done.